static/js/settings_display.js

```diff
@@ -43,33 +45,49 @@ function change_display_setting(data, $status_el, success_msg_html, sticky) {
 }
 
 function default_language_modal_post_render($container) {
-    $("#user_default_language_modal")
-        .find(".language")
-        .on("click", (e) => {
-            e.preventDefault();
-            e.stopPropagation();
-            dialog_widget.close_modal();
-
-            const $link = $(e.target).closest("a[data-code]");
-            const setting_value = $link.attr("data-code");
-            const data = {default_language: setting_value};
-
-            const new_language = $link.attr("data-name");
-            $container.find(".default_language_name").text(new_language);
-
-            change_display_setting(
-                data,
-                $container.find(".lang-time-settings-status"),
-                $t_html(
-                    {
-                        defaultMessage:
-                            "Saved. Please <z-link>reload</z-link> for the change to take effect.",
-                    },
-                    {"z-link": (content_html) => `<a class='reload_link'>${content_html}</a>`},
-                ),
-                true,
-            );
-        });
+    if ($container || page_params.is_spectator) {
+        // We don't pass $container for spectators.
+        // This also acts as an assert `$container is not None` for the else part of the syntax.
+        $("#user_default_language_modal")
+            .find(".language")
+            .on("click", (e) => {
+                e.preventDefault();
+                e.stopPropagation();
+                dialog_widget.close_modal();
+
+                const $link = $(e.target).closest("a[data-code]");
+                Cookies.set(page_params.language_cookie_name, $link.attr("data-code"));
+                window.location.reload();
```

Comment:
I am not sure how reasonable doing a reload here is vs telling user to reload it.